Google.Ads.GoogleAds/examples/HotelAds/AddPerformanceMaxForTravelGoalsCampaign.cs

```diff
@@ -0,0 +1,1037 @@
+                        FieldType = assetFieldType
+                    };
+
+                    // Adds an operation to link the Asset to the AssetGroup.
+                    mutateOperations.Add(new MutateOperation
+                    {
+                        AssetGroupAssetOperation = new AssetGroupAssetOperation
+                        {
+                            Create = assetGroupAsset
+                        }
+                    });
+                }
+
+            }
+
+            return mutateOperations;
+        }
+
+        /// <summary>
+        /// Prints the details of a MutateGoogleAdsResponse. Parses the "response" field name and
+        /// uses it to extract the new entity's name and resource name.
+        /// <param name="response">The mutate Google Ads response.</param>
+        /// </summary>
+        private void PrintResponseDetails(MutateGoogleAdsResponse response) {
+            foreach (MutateOperationResponse operationResponse in response.MutateOperationResponses)
+            {
+                string resourceName;
+
+                string entityName = operationResponse.ResponseCase.ToString();
+                // Trim the substring "Result" from the end of the entity name.
+                entityName = entityName.Remove(entityName.Length - 6);
+                switch (operationResponse.ResponseCase)
+                {
+                    case MutateOperationResponse.ResponseOneofCase.AssetResult:
+                        resourceName = operationResponse.AssetResult.ResourceName;
+                        break;
+
+                    case MutateOperationResponse.ResponseOneofCase.AssetGroupResult:
+                        resourceName = operationResponse.AssetGroupResult.ResourceName;
+                        break;
+
+                    case MutateOperationResponse.ResponseOneofCase.AssetGroupAssetResult:
+                        resourceName = operationResponse.AssetGroupAssetResult.ResourceName;
+                        break;
+
+                    case MutateOperationResponse.ResponseOneofCase.AssetSetAssetResult:
+                        resourceName = operationResponse.AssetSetAssetResult.ResourceName;
+                        break;
+
+                    case MutateOperationResponse.ResponseOneofCase.CampaignBudgetResult:
+                        resourceName = operationResponse.CampaignBudgetResult.ResourceName;
+                        break;
+
+                    case MutateOperationResponse.ResponseOneofCase.CampaignResult:
+                        resourceName = operationResponse.CampaignResult.ResourceName;
+                        break;
+
+                    default:
+                        resourceName = "<not found>";
+                        break;
+                }
+                Console.WriteLine(
+                    $"Created a(n) {entityName} with resource name: '{resourceName}'.");
+            }
+        }
+    }
+}
```

Comment:
Wow! a thousand line code example! Is there no way to make this smaller? E.g. if we provided some PMax library functionality?

Reply:
I agree it's a very big one, but the golden example in PHP is 981 lines, so I don't think there's much we can do here.
I guess it's because we want our examples to be complete and self-sufficient without requiring users to go look elsewhere for required code, so we don't want to provide helper functionalities.